### DISCUSSION TITLE
Key connector?

### DISCUSSION THREAD

**Discussion by kvvloten** [Timshel/OIDCWarden#4] [Q&A]
Hi,

Now that SSO via OIDC is working quite well, do you plan to add the key-connector functionality as well?

- Kees.

**Comment by Timshel** [comment]
Hey,

As mentioned in https://github.com/Timshel/vaultwarden/issues/41 I have no interest for the feature and nobody offered to finance its development.

Additionally, since the key connector is part of [Bitwarden license agreement](https://github.com/bitwarden/key-connector/blob/main/LICENSE.txt) it can't really be used without a license, cf:

>2.1   Commercial Module License. Subject to Your compliance with this Agreement,
Bitwarden hereby grants to You a limited, non-exclusive, non-transferable,
royalty-free license to use the Commercial Modules for the sole purposes of
internal development and internal testing, and only in a non-production
environment.

**Comment by kvvloten** [comment]
That's a quick response, thanks!

You are not interested, which is reason enough not to spend time on it :-) 

I don't understand the license argument, tough.
If you do a clean-room implementation of a specification, I cannot see what the license of another piece of software has to do with it...
This is how Linux (the Posix spec) got implemented in the nineties, why is that approach not valid for the key-connector functionality?

**Reply by Timshel** [reply]
I thought you were speaking of just integrating the Key connector that's why I mentioned the license issue.

A clean-room implementation could be done, but it's on another level in terms of work required :).
